lib/ui/platform_dispatcher.dart

```diff
@@ -308,6 +308,33 @@ class PlatformDispatcher {
     _invoke(onMetricsChanged, _onMetricsChangedZone);
   }
 
+
+  /// The framework will use this method to subscribe to platform focus changes.
+  ViewFocusChangeCallback? _onPlatformFocusChange;
+  Zone _onPlatformFocusChangeZone = Zone.root;
+  set onPlatformFocusChange(ViewFocusChangeCallback callback) {
+    assert(callback == null, 'onPlatformFocusChange was already binded');
+
+    _onPlatformFocusChange = callback;
+    _onPlatformFocusChangeZone = Zone.current;
+  }
+
+  /// The engine will use this to notify the framework a platform focus change.
+  void onPlatformFocusChange(ViewFocusChange platformFocusChange) {
+    if (_onPlatformFocusChange != null) {
+      _invoke1<ViewFocusChange>(
+        _onPlatformFocusChange,
+        _onPlatformFocusChangeZone,
+        platformFocusChange,
+      );
+    }
+  }
+
+  /// The framework will use this to notify the engine a flutter focus change.
+  void onFlutterFocusChange(ViewFocusChange flutterFocusChange) {
```

Comment:
I'd name and document this method in imperative style, as it's not a notification but a request by the framework to take/relinquish for a particular view, similar to how we have `render()` method and not `onRender`:

```
/// Requests that the engine change focus state of a view.
///
/// The target view and desired focus state are provided by the
/// `flutterFocusChange` argument.
///
/// When using this method to request focus for a particular view, any
/// other view that currently owns focus will lose focus and result in
/// the engine sending notifications through [onPlatformFocusChange]
/// callback about the previous view losing focus, and the new view
/// gaining focus.
///
/// When using this method to relinquish focus for a particular view,
/// the engine will send a notification through [onPlatformFocusChange]
/// about that view losing focus.
void requestViewFocus(ViewFocusChange flutterFocusChange);
```

Reply:
Replied below. Open to adopt this imperative style but feel the message passing approach plays better with the multiplatform nature of flutter (more in the comment below).

Reply:
Is this API also intended to be used to "unfocus" a FlutterView? In that case, the suggested name `requestFocus` would be a little misleading. What would it mean if a FlutterView would just give up focus?

Reply:
PTAL, let me know what you think about the changes I made.